drracket/drracket/private/unit.rkt

```diff
@@ -1329,8 +1329,10 @@
                  (not custodian-to-kill))
              (bell)]
             [do-kill?
-             (when custodian-to-kill
-               (custodian-shutdown-all custodian-to-kill))
+             (cond
+               ;; this transitively shuts down custodian-to-kill too
+               [ints (send ints kill-evaluation)]
+               [else (custodian-shutdown-all custodian-to-kill)])
```

Comment:
I think the else clause is a dead code, but I put it there just to be safe and consistent with previous code.

Reply:
I think it is okay to remove the else clause. Generally speaking, there aren't guards on uses of `ints` -- it should only be unset as the objects are all be created and the callback shouldn't happen until after they are all set up (because the GUI objects won't be visible until after things are all set up).

Well, I'm not 100% sure of that, but I am fairly confident that we'd fix it elsewhere if `ints` could be `#f` here.

Reply:
Done.